packages/remix-react/components.tsx

```diff
@@ -315,18 +333,18 @@ let Link = React.forwardRef<HTMLAnchorElement, RemixLinkProps>(
     let isAbsolute = typeof to === "string" && ABSOLUTE_URL_REGEX.test(to);
 
     let href = useHref(to);
-    let [shouldPrefetch, prefetchHandlers] = usePrefetchBehavior(
+    let [shouldPrefetch, ref, prefetchHandlers] = usePrefetchBehavior(
       prefetch,
       props
     );
 
     return (
       <>
         <RouterLink
-          ref={forwardedRef}
-          to={to}
           {...props}
           {...prefetchHandlers}
+          ref={mergeRefs(forwardedRef, ref)}
```

Comment:
Is there any internal react overhead in attaching a ref?  This feels potentially troublesome since we now attach a `ref` to _every_ Link, even if the user didn't forward a ref and if they're not using `prefetch="viewport"`.  Does a page with hundreds of links get immediately "slower" with no way for the user to opt out of the refs?

I took a stab at an opt-in ref approach in 44f28f3a2e514e114a818e64c3ffdbca6d5be8f0

Reply:
There's no perf hit per @jacob-ebey so going to back out that commit